tests/integration/Microsoft.Bot.Builder.Integration.Runtime.Tests/SkillConversationIdFactoryTests.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Integration.Runtime.Skills;
+using Microsoft.Bot.Builder.Skills;
+using Microsoft.Bot.Schema;
+using Xunit;
+
+namespace Microsoft.Bot.Builder.Runtime.Tests
+{
+    public class SkillConversationIdFactoryTests
+    {
+        private const string ServiceUrl = "http://testbot.com/api/messages";
+        private const string SkillId = "skill";
+
+        private string ApplicationId { get; } = Guid.NewGuid().ToString(format: "N");
+
+        private string BotId { get; } = Guid.NewGuid().ToString(format: "N");
+
+        private SkillConversationIdFactory SkillConversationIdFactory { get; } =
+            new SkillConversationIdFactory(new MemoryStorage());
+
+        [Fact]
+        public async Task SkillConversationIdFactoryHappyPath()
+        {
+            ConversationReference conversationReference = BuildConversationReference();
```

Comment:
organize your tests according to A A A (i.e. Arrange, Act and then Assert) so put the Assert at the bottom, that might mean splitting a single test up into separate targeted tests